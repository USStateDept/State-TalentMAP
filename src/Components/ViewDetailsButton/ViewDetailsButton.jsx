import PropTypes from 'prop-types';
import FontAwesome from 'react-fontawesome';
import LinkButton from '../LinkButton';

const ViewDetailsButton = ({ id }) => (
  <LinkButton toLink={`/details/${id}`} className="tm-button-alt">
    <span className="tm-button-alt-icon">
      <FontAwesome name="plus-circle" />
    </span>
    View details
  </LinkButton>
);

ViewDetailsButton.propTypes = {
  id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
};

export default ViewDetailsButton;
