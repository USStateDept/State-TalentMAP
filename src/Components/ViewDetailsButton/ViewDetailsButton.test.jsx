import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ViewDetailsButton from './ViewDetailsButton';

describe('ViewDetailsButtonComponent', () => {
  it('is defined', () => {
    const wrapper = shallow(
      <ViewDetailsButton id="1" />,
    );
    expect(wrapper).toBeDefined();
  });

  it('can take props', () => {
    const wrapper = shallow(
      <ViewDetailsButton id="1" />,
    );
    expect(wrapper.props().toLink).toBe('/details/1');
  });

  it('matches snapshot', () => {
    const wrapper = shallow(
      <ViewDetailsButton id="1" />,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
