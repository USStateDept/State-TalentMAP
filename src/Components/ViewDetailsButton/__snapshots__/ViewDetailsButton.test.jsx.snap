// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewDetailsButtonComponent matches snapshot 1`] = `
<LinkButton
  className="tm-button-alt"
  isExternal={false}
  toLink="/details/1"
  useDefaultClass={true}
>
  <span
    className="tm-button-alt-icon"
  >
    <FontAwesome
      name="plus-circle"
    />
  </span>
  View details
</LinkButton>
`;
